src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,94 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPASITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private static final int GROWTH_FACTOR = 2;
+    private static final int DEFAULT_SIZE = 0;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPASITY];
+        threshold = (int) (DEFAULT_CAPASITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size == threshold) {
+            resize();
+        }
+        int index = getIndex(key);
+        if (table[index] == null) {
+            table[index] = new Node<>(key, value);
+            size++;
+            return;
+        } else {
+            Node<K, V> tableNode = table[index];
+            while (tableNode != null) {
+                if (Objects.equals(tableNode.key, key)) {
+                    tableNode.value = value;
+                    return;
+                }
+                tableNode = tableNode.next;
+            }
+        }
+        Node<K, V> newNode = new Node<>(key, value);
+        newNode.next = table[index];
+        table[index] = newNode;
+        size++;
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> tableNode = table[getIndex(key)];
+        while (tableNode != null && !Objects.equals(tableNode.key, key)) {
+            tableNode = tableNode.next;
+        }
+        if (tableNode == null) {
+            return null;
+        }
+        return tableNode.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        Node<K, V>[] newTable = new Node[table.length * GROWTH_FACTOR];
+        threshold = (int) (newTable.length * DEFAULT_LOAD_FACTOR);
+        this.size = DEFAULT_SIZE;
+        copyArray(table);
+    }
+
+    private void copyArray(Node<K, V>[] oldTable) {
```

Comment:
Bad name. it's not a copy array. This method transfer all node from one hash map to another.